nixos/modules/services/misc/matrix-conduit.nix

```diff
@@ -0,0 +1,168 @@
+              description = "The port Conduit will be running on. You need to set up a reverse proxy in your web server (e.g. apache or nginx), so all requests to /_matrix on port 443 and 8448 will be forwarded to the Conduit instance running on this port";
+            };
+            global.max_request_size = mkOption {
+              type = types.ints.positive;
+              default = 20000000;
+              description = "Max request size in bytes. Don't forget to also change it in the proxy.";
+            };
+            global.allow_registration = mkOption {
+              type = types.bool;
+              default = false;
+              description = "Whether new users can register on this server.";
+            };
+            global.allow_encryption = mkOption {
+              type = types.bool;
+              default = false;
+              description = "Whether new encrypted rooms can be created. Note: existing rooms will continue to work.";
+            };
+            global.allow_federation = mkOption {
+              type = types.bool;
+              default = false;
+              description = ''
+                Whether this server federates with other servers.
+                Upstream documentation contains the following warning:
+                You should not join Matrix rooms without asking the admins first.
+                We do not know whether Conduit is safe for general use yet, so you
+                should assume there is some chance that it breaks rooms permanently
+                for all participating users. We are not aware of such a bug today,
+                but we would like to do more testing.
+              '';
+            };
+            global.allow_jaeger = mkOption {
+              type = types.bool;
+              default = false;
+              description = "Whether jaeger to support monitoring and troubleshooting through jaeger is enabled on this server.";
+            };
+            global.trusted_servers = mkOption {
+              type = types.listOf types.str;
+              default = [ "matrix.org" ];
+              description = "Servers trusted with signing server keys.";
+            };
+
+            global.address = mkOption {
+              type = types.str;
+              default = "127.0.0.1";
+              description = "Address to listen on for connections by the reverse proxy/tls terminator.";
+            };
+          };
+        };
+        default = {};
+        description = ''
+            generates the conduit.toml configuration file. Refer to
+            <link xlink:href="https://gitlab.com/famedly/conduit/-/blob/master/conduit-example.toml"/>
+            for details on supported values.
+            Note that database_path should not be set because the services reliance on systemd StateDir.
+        '';
+      };
+    };
+
+
+    config = mkIf cfg.enable {
+      services.conduit.settings = {
+        global.database_path = "/var/lib/matrix-conduit/conduit.db";
```

Comment:
Can we set it as default above instead? This would also allow adding the `description` to the place where it makes most sense instead of burring the warning (why are we warning? What are the perquisites for setting it?) in the generic settings description.

Reply:
And since you cannot really give a dynamic user permissions to something outside of its defined directories I feel it would only be reasonable to set it `readOnly`, else we would require allocating static uid/gid.

Yep, that limits you somewhat in regard to what volume you can store your state on, but I think `DynamicUser` is a sensible choice made in lots of modules these days. The limitation with storing these directories on another volume should probably be carried upstream.